lib/bloc/podcast_clip/podcast_clip_bloc.dart

```diff
@@ -0,0 +1,238 @@
+    }
+
+    _clipDetailsBehaviourSubject.add(clipDetails.copy(
+        clipDuration: incrementedClipDuration,
+        endTimeStamp: incrementedClipEndTime));
+  }
+
+  decrementCount() async {
+    var clipDetails = await getCurrentPodcastClipDetails();
+    int currentCLipDuration = clipDetails.clipDuration;
+
+    //return if duration needs to be greator than 10 seconds
+    if (currentCLipDuration <= _initialSeconds) {
+      return;
+    }
+    int decrementedClipDuration = 0;
+    Duration decrementedClipEndTime = Duration(seconds: 0);
+
+    if (currentCLipDuration % 10 != 0) {
+      decrementedClipDuration = currentCLipDuration - currentCLipDuration % 10;
+      decrementedClipEndTime = Duration(
+          seconds:
+              clipDetails.endTimeStamp.inSeconds - currentCLipDuration % 10);
+    } else {
+      decrementedClipDuration = currentCLipDuration - _initialSeconds;
+      decrementedClipEndTime =
+          clipDetails.endTimeStamp - Duration(seconds: _initialSeconds);
+    }
+
+    _clipDetailsBehaviourSubject.add(clipDetails.copy(
+        clipDuration: decrementedClipDuration,
+        endTimeStamp: decrementedClipEndTime));
+  }
+
+  Future<String> _getAudioClipPath(
+      {PodcastClipDetailsModel clipDetails}) async {
+    var clipDetails = await getCurrentPodcastClipDetails();
+
+    String episodeUrl = clipDetails.episodeDetails.contentUrl;
+    String audioClipPath = await initClipDirectory(isVideoClip: false);
+    audioClipPath = audioClipPath + '/clip.mp3';
+
+    //Getting the raw audio clip
+    String clippedAudioCommand =
+        '-i "$episodeUrl" -ss ${clipDetails.startTimeStamp.inSeconds} -to ${clipDetails.endTimeStamp.inSeconds}  -acodec copy $audioClipPath';
+
+    FFmpegSession ffpegSessionDetails =
+        await FFmpegKit.execute(clippedAudioCommand);
+
+    final returnCode = await ffpegSessionDetails.getReturnCode();
+    if (ReturnCode.isSuccess(returnCode)) {
+      return audioClipPath;
+    } else {
+      return null;
+    }
+  }
+
+  getVideoClipPath({String audioClipPath, String episodeImagepath}) async {
+    String videoClipPath = await initClipDirectory(isVideoClip: true);
+    videoClipPath = videoClipPath + '/VideoClip.mp4';
+
+    String clippedVideoCommand =
+        '-i  $audioClipPath -i $episodeImagepath -filter_complex "[0:a]showwaves=colors=0xffffff@0.9:mode=p2p,format=yuva420p[v];[1:v]scale=2120:2560[bg];[bg][v]overlay=(main_w-overlay_w)/2:H*0.8[outv]" -map "[outv]" -map 0:a -c:v libx264 -c:a copy $videoClipPath';
+
+    FFmpegSession ffpegSessionDetails =
+        await FFmpegKit.execute(clippedVideoCommand);
+
+    String logs = await ffpegSessionDetails.getLogsAsString();
+    log('cliplogs $logs');
+    final returnCode = await ffpegSessionDetails.getReturnCode();
+    if (ReturnCode.isSuccess(returnCode)) {
+      return videoClipPath;
+    } else {
+      return null;
+    }
+  }
+
+  Future<bool> isEpisodeClipable() async {
```

Comment:
Why do we need this method? It seems like the code ensures that the clipDetails always maintains clip-able episode.

Reply:
This method checks if there is time to create a clip of minimum 10 secs. If not, the bottom sheet won't be shown to the user.